src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToBuildASelfContainedApp.cs

```diff
@@ -383,6 +385,75 @@ public void It_can_publish_runtime_specific_apps_with_library_dependencies_self_
             publishCommand.Execute(new[] { "-property:SelfContained=true", "-property:_CommandLineDefinedSelfContained=true", $"-property:RuntimeIdentifier={rid}", "-property:_CommandLineDefinedRuntimeIdentifier=true" }).Should().Pass().And.NotHaveStdOutContaining("warning");
         }
 
+        [Theory]
+        [InlineData("net7.0")]
+        [InlineData("net8.0")]
+        public void It_does_or_doesnt_imply_SelfContained_based_on_RuntimeIdentifier_and_TargetFramework(string targetFramework)
+        {
+            var runtimeIdentifier = EnvironmentInfo.GetCompatibleRid(targetFramework);
+            bool resultShouldBeSelfContained = targetFramework == "net7.0" ? true : false;
+            var testProject = new TestProject("MainProject")
+            {
+                TargetFrameworks = targetFramework,
+                IsExe = true
+            };
+
+            testProject.RecordProperties("SelfContained");
+            testProject.AdditionalProperties["RuntimeIdentifier"] = runtimeIdentifier;
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: targetFramework);
+            new DotnetBuildCommand(Log)
+                .WithWorkingDirectory(Path.Combine(testAsset.Path, "MainProject"))
+                .Execute()
+                .Should()
+                .Pass();
+
+            var properties = testProject.GetPropertyValues(testAsset.TestRoot, targetFramework: targetFramework);
+            Assert.Equal(bool.Parse(properties["SelfContained"]), resultShouldBeSelfContained);
+        }
+
+        [Theory]
+        [InlineData("net7.0", true)]
+        [InlineData("net7.0", false)]
+        [InlineData("net8.0", false)]
+        public void It_does_or_doesnt_warn_based_on_SelfContained_and_TargetFramework_breaking_RID_change(string targetFramework, bool defineSelfContained)
+        {
+            var runtimeIdentifier = EnvironmentInfo.GetCompatibleRid(targetFramework);
+            var testAsset = _testAssetsManager
+                .CopyTestAsset("HelloWorld", identifier: targetFramework + defineSelfContained.ToString())
+                .WithSource()
+                .WithTargetFramework(targetFramework)
+                .WithProjectChanges(project =>
+                {
+                    var ns = project.Root.Name.Namespace;
+                    var propertyGroup = project.Root.Elements(ns + "PropertyGroup").First();
+                    propertyGroup.Add(new XElement(ns + "RuntimeIdentifier", runtimeIdentifier));
+                    propertyGroup.Add(new XElement(ns + "SelfContained", defineSelfContained ? "true" : ""));
+                });
+
+            var buildCommand = new DotnetBuildCommand(Log);
+            var commandResult = buildCommand
+                .WithWorkingDirectory(testAsset.Path)
+                .Execute("/bl");
```

Comment:
I think we should avoid producing binlogs during CI test runs, they just take up extra space (and it would be a lot of extra space if all the tests were doing it).  When I'm debugging locally I'll add the `/bl` parameter if needed, and sometimes I'll accidentally check it in.  But I think we should avoid doing it intentionally and remove it when we notice it.

Reply:
Thanks for explaining. I agree.